src/drawing/DrawingCreator.tsx

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import classNames from 'classnames';
+import noop from 'lodash/noop';
+import * as uuid from 'uuid';
+import DrawingPath, { DrawingPathRef, getPathCommands } from './DrawingPath';
+import DrawingSVG, { DrawingSVGRef } from './DrawingSVG';
+import PointerCapture, { Status as DrawingStatus } from '../components/PointerCapture';
+import { getStrokeWidths } from './DrawingPathGroup';
+import { PathGroup, Position, Stroke } from '../@types';
+import './DrawingCreator.scss';
+
+type Props = {
+    className?: string;
+    onAbort?: () => void;
+    onStart: () => void;
+    onStop: (pathGroup: PathGroup) => void;
+    stroke?: Stroke;
+};
+
+const defaultStroke = {
+    color: '#4826c2',
+    size: 4,
+};
+
+export default function DrawingCreator({
+    className,
+    onAbort = noop,
+    onStart,
+    onStop,
+    stroke = defaultStroke,
+}: Props): JSX.Element {
+    const [drawingStatus, setDrawingStatus] = React.useState<DrawingStatus>(DrawingStatus.init);
+    const capturedPathRef = React.useRef<Array<Position>>([]);
+    const creatorElRef = React.useRef<HTMLDivElement>(null);
+    const drawingDirtyRef = React.useRef<boolean>(false);
+    const drawingPathRef = React.useRef<DrawingPathRef>(null);
+    const drawingSVGRef = React.useRef<DrawingSVGRef>(null);
+    const renderHandleRef = React.useRef<number | null>(null);
+
+    const getPathGroup = (): PathGroup | null => {
+        const { current: creatorEl } = creatorElRef;
+        const { current: points } = capturedPathRef;
+
+        if (!creatorEl || points.length <= 1) {
+            return null;
+        }
+
+        return {
+            clientId: uuid.v4(),
+            paths: [{ clientId: uuid.v4(), points }],
+            stroke,
+        };
+    };
+    const getPosition = (x: number, y: number): [number, number] => {
+        const { current: creatorEl } = creatorElRef;
+
+        if (!creatorEl) {
+            return [x, y];
+        }
+
+        // Calculate the new position based on the mouse position less the page offset
+        // and convert to a percentage of the page
+        const { height, left, top, width } = creatorEl.getBoundingClientRect();
+        return [((x - left) / width) * 100, ((y - top) / height) * 100];
```

Comment:
The RegionCreator returns pixel values for this method. Can we keep the behavior consistent?

Reply:
I can do that, but the tradeoff of that is for each animation frame render we'll need to recalculate the % values for all the points in the current path and then generate the path commands for the SVG whereas if we do this at the time of storing the points in the `capturedPathRef` then we simply generate the path commands. Seems like we'd want to lean toward the path of less processing during each animation frame.